adapters/p2p2core/transaction.go

```diff
@@ -71,13 +71,34 @@ func AdaptTransaction(t *spec.Transaction, network *utils.Network) core.Transact
 		return declareTx
 	case *spec.Transaction_DeclareV3_:
 		tx := t.GetDeclareV3()
+
+		nDAMode, err := strconv.ParseUint(tx.GetNonceDomain(), 10, 32)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to convert Nonce DA mode: %v to uint32", tx.GetNonceDomain()))
+		}
+
+		fDAMode, err := strconv.ParseUint(tx.GetFeeDomain(), 10, 32)
+		if err != nil {
+			panic(fmt.Sprintf("Failed to convert Fee DA mode: %v to uint32", tx.GetFeeDomain()))
+		}
```

Comment:
I think we shouldn't use panic here because a malicious peer can send any information in these fields, moreover since they are of string type it should be pretty easy to trigger error

Reply:
I decided to add those panic because I didn't want `Adapt` to return an error. Also, since some of the code is deprecated I didn't want to spend more time trying to do something proper. 

For now only Juno is going to be sending those messages.